python/ray/train/base_trainer.py

```diff
@@ -362,6 +366,21 @@ def fit(self) -> Result:
             raise TrainingFailedError from e
         return result
 
+    def _extract_fields_for_tuner_param_space(self) -> Dict:
+        """Extracts fields to be included in `Tuner.param_space`.
+
+        This is needed to leverage the full logging/integration offerings from Tune.
+        For example, `param_space` is logged automatically to wandb integration.
+
+        Returns:
+            A dictionary that should be passed to Tuner.param_space.
+        """
+        result = {}
+        for key in self._param_dict.keys():
+            if key in ["train_loop_config"]:
+                result[key] = copy.deepcopy(self._param_dict[key])
+        return result
```

Comment:
can this just be `self_param_dict.get("train_loop_config", {})`?

Reply:
Ah i see. I would prefer if `train_loop_config` is not in the picture, we should not upload something like "train_loop_config=None or {}" to wandb. 

I inverted the inner and outer loop to make it more efficient.

Also added a test to assert that train_loop_config won't always be uploaded.

Reply:
> we should not upload something like "train_loop_config=None or {}" to wandb.

Not sure if I'm misunderstanding this comment, but doesn't this happen with current implementation?

Like, if your trainer looks like

```
TorchTrainer(
    ...
    train_loop_config=None  # Or `train_loop_config` isn't specified
)
```

Then `"train_loop_config" in self._param_space` and `_extract_fields_for_tuner_param_space` returns `{"train_loop_config": None}`?

Reply:
hmm I am confused - maybe I am missing something here.

For
```
TorchTrainer(
    ...
    train_loop_config=None 
)
```
it will be uploaded as None. But I don't think that's a big deal as the user specified it (for no good reason imo)

However, if it's
```
TorchTrainer(
...
)
```
Then it won't show up. It's not in `self._param_space`.

I printed `self._param_space` for reference:

```
{'train_loop_per_worker': <function test_trainer_wandb_integration.<locals>.train_func at 0x7fdb3b108a70>, 'scaling_config': ScalingConfig(num_workers=2), 'run_config': RunConfig(callbacks=[<ray.air.tests.mocked_wandb_integration.WandbTestExperimentLogger object at 0x7fdb3b103fd0>])}
```

Also see more in the added unit test.

Reply:
You're right, I misread the `__new__` implementation; `train_loop_config` isn't included in `self._param_space` if it isn't explicitly specified.